modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java

```diff
@@ -1,13 +1,13 @@
 package org.testcontainers.containers;
 
-import com.google.common.collect.Sets;
 import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
 import org.testcontainers.utility.DockerImageName;
 
 import java.time.Duration;
+import java.util.HashSet;
 import java.util.Set;
 
-public class ClickHouseContainer extends JdbcDatabaseContainer {
+public class ClickHouseContainer<SELF extends ClickHouseContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
```

Comment:
No need to have self-typing, also breaks backward compatibility:

```suggestion
public class ClickHouseContainer extends JdbcDatabaseContainer<ClickHouseContainer> {
```